extensions/agroal/deployment/src/main/resources/dev-ui/qwc-agroal-datasource.js

```diff
@@ -0,0 +1,339 @@
+                            <vaadin-grid-sort-column path="columnSize" auto-width resizable></vaadin-grid-sort-column>
+                            <vaadin-grid-sort-column path="nullable" auto-width resizable></vaadin-grid-sort-column>
+                        </vaadin-grid>`;
+        }
+    }
+    
+    _renderSqlInput(){
+        return html`<div class="sqlInput">
+                        <qui-code-block @shiftEnter=${this._shiftEnterPressed} content="${this._currentSQL}" id="sql" mode="sql" theme="dark" value='${this._currentSQL}' editable></qui-code-block>
+                        <vaadin-icon class="sqlInputButton" title="Clear" icon="font-awesome-solid:broom" @click=${this._clearSqlInput}></vaadin-icon>
+                        <vaadin-icon class="sqlInputButton" title="Run" icon="font-awesome-solid:person-running" @click=${this._executeClicked}></vaadin-icon>
+                    </div>`;
+    }
+    
+    _columnNameRenderer(col){
+        if(this._selectedTable.primaryKeys.includes(col.columnName)){
+            return html`${col.columnName} <vaadin-icon class="pkicon" icon="font-awesome-solid:key"></vaadin-icon>`;
+        }else{
+            return html`${col.columnName}`;
+        }
+    }
+    
+    _watch(){
+        this._isWatching = true;
+        this._watchId = setInterval(() => {
+            this.hotReload();
+        }, 3000);
+    }
+    
+    _unwatch(){
+        this._isWatching = false;
+        clearInterval(this._watchId);
+        this._watchId = null;
+    }
+    
+    _onDataSourceChanged(event) {
+        const selectedValue = event.detail.value;
+        if(selectedValue in this._dataSources){
+            this._selectedDataSource = this._dataSources[selectedValue];
+            this._fetchTableDefinitions();
+        }
+    }
+    
+    _onTableChanged(event){
+        this._selectedTableIndex = event.detail.value;
+        this._selectedTable = this._tables[this._selectedTableIndex];
+        this._clearSqlInput();
+    }
+    
+    _fetchAllData(){
+        this.jsonRpc.executeSQL({datasource:this._selectedDataSource.name, sql:this._currentSQL}).then(jsonRpcResponse => {
+            if(jsonRpcResponse.result.error){
+                notifier.showErrorMessage(jsonRpcResponse.result.error);
+            }else{
+                this._currentDataSet = jsonRpcResponse.result;
+            }
+        });
+    }
+    
+    _fetchTableDefinitions() {
+        this.jsonRpc.getTables({datasource:this._selectedDataSource.name}).then(jsonRpcResponse => {
+            this._tables = jsonRpcResponse.result;
+        });
+    }
+    
+    _executeClicked(event){
+        let newValue = this.shadowRoot.getElementById('sql').getAttribute('value');
+        this._executeSQL(newValue);
+    }
+    
+    _clearSqlInput(){
+        this._executeSQL("select * from " + this._selectedTable.tableName);
+        this._fetchAllData();
```

Comment:
How will it scale when you have a lot of data?
That could happen even in Dev UI as you might have an already populated database. At least, that's how we worked at my previous company once the initial development was over, we were working daily with a fully populated dataset coming from the production.

Reply:
I limit on the server to max 100. The reason I am not doing it here (in Sql) is the SQL is different per vendor. So I just do it in the result set

Reply:
Then the UI should reflect this somehow, or?

Reply:
Related: https://github.com/quarkusio/quarkus/pull/43618#discussion_r1782441706